hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/JpaDeploymentManagement.java

```diff
@@ -883,66 +915,44 @@ public long countMessagesByActionStatusId(final long actionStatusId) {
         return entityManager.createQuery(countMsgQuery).getSingleResult();
     }
 
-    @Override
-    public Page<ActionStatus> findActionStatusAll(final Pageable pageable) {
-        return JpaManagementHelper.findAllWithCountBySpec(actionStatusRepository, pageable, null);
-    }
-
-    @Override
-    public long countActionStatusAll() {
-        return actionStatusRepository.count();
-    }
-
     @Override
     public long countActionsAll() {
         return actionRepository.count();
     }
 
     @Override
     public long countActions(final String rsqlParam) {
-        final List<Specification<JpaAction>> specList = Arrays.asList(
+        final List<Specification<JpaAction>> specList = List.of(
                 RSQLUtility.buildRsqlSpecification(rsqlParam, ActionFields.class, virtualPropertyReplacer, database));
         return JpaManagementHelper.countBySpec(actionRepository, specList);
     }
 
-    @Override
-    public long countActionsByDistributionSetIdAndActiveIsTrue(final Long distributionSet) {
-        return actionRepository.countByDistributionSetIdAndActiveIsTrue(distributionSet);
-    }
-
-    @Override
-    public long countActionsByDistributionSetIdAndActiveIsTrueAndStatusIsNot(final Long distributionSet,
-            final Status status) {
-        return actionRepository.countByDistributionSetIdAndActiveIsTrueAndStatusIsNot(distributionSet, status);
-    }
-
-    @Override
-    public Slice<Action> findActionsByDistributionSet(final Pageable pageable, final long dsId) {
-        throwExceptionIfDistributionSetDoesNotExist(dsId);
-        return actionRepository.findByDistributionSetId(pageable, dsId);
-    }
-
+    // TODO - return via Mgmt API all actions (event for targets the use has no access - check if should and could
+    // be limited
```

Comment:
I'm not sure how effective an action AccessController could be implemented.
Anyway these methods will just return actions and its implicitly loaded JPA entities. 
however max read access to probably not sensitive information will be achieved.
Maybe at some later point we could add restriction on getters level - so only action will be returned but Action#getTarget will fail.
For now, I think this could stay like it is. What do you think?